tests/testthat/test-differentialAbundance.R

```diff
@@ -431,3 +431,54 @@ test_that("differentialAbundance catches deseq errors", {
 
 
 })
+
+test_that("differentialAbundance handles non-integer data",{
+    df <- testOTU
+  counts <- df[, -c("entity.SampleID")]*1000
+  counts[ ,entity.SampleID:= df$entity.SampleID]
+  nSamples <- dim(df)[1]
+  testSampleMetadata <- data.frame(list(
+    "entity.SampleID" = df[["entity.SampleID"]],
+    "entity.binA" = rep(c("binA_a", "binA_b"), nSamples/2, replace=T),
+    "entity.cat3" = rep(paste0("cat3_", letters[1:3]), nSamples/3, replace=T),
+    "entity.cat4" = rep(paste0("cat4_", letters[1:4]), nSamples/4, replace=T),
+    "entity.contA" = rnorm(nSamples, sd=5)
+    ))
+
+
+  testData <- microbiomeComputations::AbundanceData(
+              data = counts,
```

Comment:
this can just be testOTU, no need to make `counts` anymore

Reply:
I'm going to expand the tests a bit yet